checks/default.nix

```diff
@@ -1,18 +1,42 @@
-{ pkgs, modules }:
+{ self, ... }:
 {
-  # We mostly check that the module evaluates.
-  developer = pkgs.testers.nixosTest {
-    name = "developer";
+  perSystem =
+    { pkgs, ... }:
+    {
+      # We mostly check that the module evaluates.
+      checks.developer = pkgs.testers.nixosTest {
+        name = "developer";
 
-    nodes.machine = {
-      imports = [ modules.developer ];
+        nodes.machine = {
+          imports = [ self.nixosModules.developer ];
 
-      ctrl-os.developer.enable = true;
-    };
+          ctrl-os.developer.enable = true;
+        };
+
+        testScript = ''
+          start_all()
+          machine.wait_for_unit("multi-user.target")
+        '';
+      };
 
-    testScript = ''
-      start_all()
-      machine.wait_for_unit("multi-user.target")
-    '';
-  };
+      pre-commit = {
+        check.enable = true;
```

Comment:
Maybe `check.enable = lib.take 2 (lib.splitString "." lib.version) == [ "25" "11" ];` does the trick with the CI?